server.go

```diff
@@ -0,0 +1,137 @@
+package sortthread
+
+import (
+	"errors"
+
+	"github.com/emersion/go-imap"
+	"github.com/emersion/go-imap/server"
+)
+
+var ErrUnsupportedBackend = errors.New("sortthread: backend not supported")
+
+type SortMailbox interface {
```

Comment:
Should we embed `imap.Mailbox` here?

Reply:
backend.Mailbox, yes, we should as I already answered in "Merge small extensions" PR.